modin/pandas/base.py

```diff
@@ -3440,7 +3440,16 @@ def __array__(self, dtype=None):
         arr : np.ndarray
             NumPy representation of Modin object.
         """
-        arr = self.to_numpy(dtype)
+        from modin.config import ExperimentalNumPyAPI
+
+        exp_numpy_api = ExperimentalNumPyAPI.get()
+        try:
+            if exp_numpy_api:
+                ExperimentalNumPyAPI.put(False)
+            arr = self.to_numpy(dtype)
+        finally:
+            if exp_numpy_api:
+                ExperimentalNumPyAPI.put(True)
```

Comment:
I don't like this juggling... I propose we introduce a non-standard `_to_bare_numpy()` (or something similar named) API method which would always return non-wrapped Numpy arrays instead of all this configuration trickery.

Reply:
Some of Modin's code itself is tied to the fact that this function returns a regular numpy, and all third-party libraries that check if `__array__` method exists, without checking the type of the object.

I think that we have to enforce the contract of `__array__` function in some way.

P.S. `numpy.array(modin_array)` will failed if we don't make the changes.

Reply:
How this relates to what I said? :) What I'm saying is that `__array__` implementation should be different, so that it doesn't set-unset the configuration variable.

See `modin/utils.py::to_numpy()` implementation.

Reply:
misunderstood you